skfp/preprocessing.py

```diff
@@ -64,3 +65,67 @@ def transform(self, X):
             )
             for x in X
         ]
+
+
+class ConformerGenerator:
+    def __init__(self, max_conf_gen_attempts: int = 1000, random_state: int = 0):
+        self.max_conf_gen_attempts = max_conf_gen_attempts
+        self.random_state = random_state
+
+    def fit(self, X, y=None, **fit_params):
+        return self
+
+    def fit_transform(self, X, y=None, **fit_params):
+        return self.transform(X)
+
+    def transform(self, X):
+        # adding hydrogens is recommended for conformer generation
+        X = [AddHs(mol) for mol in X]
+
+        embed_params = ETKDGv3()
+        embed_params.useSmallRingTorsions = True
+        embed_params.randomSeed = self.random_state
+
+        conf_ids = []
+        for mol in X:
+            conf_id = -1
+
+            try:
+                # basic attempt
+                conf_id = EmbedMolecule(mol, embed_params)
+            except ValueError:
+                pass
+
+            if conf_id == -1:
+                try:
+                    # more tries
+                    embed_params.maxIterations = self.max_conf_gen_attempts
+                    embed_params.useRandomCoords = True
+                    conf_id = EmbedMolecule(mol, embed_params)
+                except ValueError:
+                    pass
+
+            if conf_id == -1:
+                try:
+                    # turn off conditions
+                    embed_params.enforceChirality = False
+                    embed_params.ignoreSmoothingFailures = True
+                    conf_id = EmbedMolecule(mol, embed_params)
+                except ValueError:
+                    pass
```

Comment:
Wouldn't it make sense to replace these 2 blocks with one if statement and try/except? I don't think it would impact readability

Reply:
I tried that first, but it results in more indents and was less readable in my opinion. I further simplified this file now